remote-view/shared/src/main/scala/parsley/debug/RemoteView.scala

```diff
@@ -10,130 +10,155 @@ import scala.util.Failure
-  private [debug] final val DefaultBreakpointSkip = -1
-
-  /**
-   * Send the debug tree and input to the port and address specified in the 
-   * object construction. 
-   *
-   * @param input The input source.
-   * @param tree The debug tree.
-   */
-  override private [debug] def render(input: => String, tree: => DebugTree): Unit = {
-    // Return value of the renderWithTimeout function not needed for a regular parse
-    val _ = renderWithTimeout(input, tree, ResponseTimeout)
-  }
-  /**
-    * Send the debug tree and input to the port and address specified in the
-    * object construction.
-    * 
-    * This function will block and wait for a response from the remote view.
-    * This is to allow breakpoints to halt Parsley parsing and wait for a 
-    * number of breakpoints to skip.
-    * 
-    * The number of breakpoints to skip represents:
-    *   n == 0  : Move through the current breakpoint and halt on the next.
-    *   n >= 1  : Move through the current breakpoint and skip the next n breakpoints.
-    *   n <= -1 : Stop the parser and exit the program.
-    *
-    * @param input The input source.
-    * @param tree The debug tree.
-    * @param timeout The maximal timeout of the connection.
-    * @param isDebuggable If the instance is a debuggable instance.
-    * 
-    * @return The number of breakpoints to skip after this breakpoint exits.
-    */
-  override private [debug] def renderWait(input: => String, tree: => DebugTree): Int =
-    renderWithTimeout(input, tree, BreakpointTimeout, isDebuggable = true).flatMap(_.skipBreakpoint).getOrElse(DefaultBreakpointSkip)
-
-  private [debug] def renderWithTimeout(input: => String, tree: => DebugTree, timeout: FiniteDuration, isDebuggable: Boolean = false): Option[RemoteViewResponse] = {
-    // JSON formatted payload for post request
-    val payload: String = DebugTreeSerialiser.toJSON(input, tree, isDebuggable)
+* local host (by default) or to a specified IP address. This enables all of the debug tree parsing, serving and 
+* graphics to be done in a separate process, and served reactively to a client.
+* 
+* This module is part of the Debugging Interactively in parsLey Library (DILL) project,
+* (https://github.com/j-mie6/parsley-debug-app).
+* 
+* RemoteView uses the STTP library to create HTTP requests to a specified IP address, over a specified port.
+* The request is formatted using the upickle JSON formatting library, it is being used over other
+* libraries like circe for its improved speed over large data structures.
+*/
+sealed trait RemoteView extends DebugView.Reusable with DebugView.Pauseable with DebugView.Manageable {
+    protected val port: Int
+    protected val address: String
     
-    // Send POST
-    println("Sending Debug Tree to Server")
-
-    // Implicit JSON deserialiser
-    implicit val responsePayloadRW: RW[RemoteViewResponse] = macroRW[RemoteViewResponse]
+    // Printing helpers
+    private [debug] final val TextToRed    = "\u001b[31m"
```

Comment:
`Console.Red`? There is something for this, check in the debug instrs for vanilla parsley